lib/rails_failover/active_record/handler.rb

```diff
@@ -64,7 +64,14 @@ def initiate_fallback_to_primary
                 spec_name,
                 role: handler_key,
               )
-            connection_active = connection.active?
+
```

Comment:
Ha, strange that the behavior is the same between Rails 7.1 and 7.2. The specs weren’t catching this with Rails 7.1. Nice catch! :+1: 

I fixed this in my PR for Rails 7.2:
https://github.com/discourse/rails_failover/pull/70/files#diff-f5a4f2625d725f91fc45b4610ee04e863e06f2c0d2ec51bf7404dff788b06315

We should be able to simply call `.tap(&:verify!)` on the retrieved connection (the method exists in Rails 7.0 too) to make it active if it’s not. Then no need to execute a fake query, etc :slightly_smiling_face: